packages/react-components/react-swatch-picker-preview/docs/Spec.md

```diff
@@ -1,5 +1,262 @@
+| Color cell                                                   | ColorPickerGridCell | ColorSwatch  | ⚠️        |
+| Image cell                                                   |                     | ImageSwatch  | ❌        |
+
+## Sample Code
+
+```jsx
+<SwatchPicker
+  layout="grid"
+  swatch={color}
+  onChange={(\_, data) => setSwatch(data.value)}
+  aria-labelledby="colors"
+  name="color"
+  style={{ gridTemplateColumns: `repeat(4, 30px)` }} >
+    <ColorSwatch id={0} swatch="red" />
+    <ColorSwatch id={1} swatch="rgb(189, 255, 104)" />
+    <ColorSwatch id={2} swatch="#f09" />
+    <ColorSwatch id={3} swatch="#ad5" />
+    <ColorSwatch id={4} swatch="magenta" />
+    <ImageSwatch id={5} swatch="./path/image.png"/>
+    <ImageSwatch id={6} swatch="./path/image1.png"/>
+    <ImageSwatch id={7} swatch="./path/image2.png"/>
+    <ImageSwatch id={8} swatch="./path/image3.png"/>
+</SwatchPicker>
+```
+
+## Variants
+
+### Layout variants
+
+- Grid
+- Row
+
+Grid starts from 4 swatches.
+SwatchPiker is responsive by default.
+Grid spacing can be customized using CSS `grid-template` props.
+
+### Swatch Variants
+
+- Color
+- Gradient
+- Image
+- Pattern / texture
+
+### Shapes
+
+- `square` (default)
+- `circular`
+- `rounded`
+
+Border radius for rounded shape can be customized via CSS.
+
+### Size
+
+- `extraSmall`: 20px
+- `small`: 24px
+- `medium` (default): 28px
+- `large`: 32px
+
+Custom size can be set by overriding `width` and `height` of the ColorSwatch or ImageSwatch.
+
+### Density/Gap
+
+Gap is 2px by default.
+Horizontal and vertical density can be changed via CSS.
+
+### States
+
+- `hover`
+- `selected`
+- `focused`
+- `pressed`
+
+## API
+
+## Structure
+
+### Components
+
+| Component    | Purpose                                                |
+| ------------ | ------------------------------------------------------ |
+| SwatchPicker | Group which can represent swatches as a row or a grid. |
+| ColorSwatch  | Swatch for a color                                     |
+| ImageSwatch  | Swatch for an image, texture or a pattern              |
+
+## SwatchPicker component
+
+#### Anatomy
+
+![visual anatomy of the SwatchPicker component](./assets/todo.png)
+
+#### DOM
+
+```HTML
+<div role="radiogroup" aria-labelledby="colors">
+  {children}
+</div>
+```
+
+#### ColorPicker structure
+
+- root `div` element
+
+#### API
+
+| Property | Values                                   | Default  | Purpose                         |
+| -------- | ---------------------------------------- | -------- | ------------------------------- |
+| shape    | `square`, `circular`, `rounded`          | `square` | Sets shape                      |
+| size     | `extraSmall`, `small`, `medium`, `large` | `medium` | Defines size of the Swatch cell |
+| id       | number                                   |          | Sets swatch id                  |
+| layout   | `row`, `grid`                            | `grid`   | Sets layout of the SwatchPicker |
```

Comment:
If we have a separate component `SwatchPickerRow`, we won't need a `layout` prop

Reply:
No need Row component. We'll use css grid